### PR TITLE
Add Model runner compatibility test

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/numenta-apps/issues/436.

### CODE REVIEW

**Comment by chetan51** [comment]
@marionleborgne @oxtopus

File: unicorn/scripts/convert_data.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        outfile.write("\n")
+
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser()
+  parser.add_argument('data', metavar='/path/to/data.csv', type=str)
+  parser.add_argument('output', metavar='/path/to/output.json', type=str)
+  parser.add_argument('-n', '--num_header_rows', type=int, default=3)
+  parser.add_argument('-t', '--timestamp_format', type=str, default='%Y-%m-%d %H:%M:%S')
```
Comment (marionleborgne) [inline unicorn/scripts/convert_data.py]: more than 80 chars per line

File: unicorn/scripts/convert_data.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import argparse
+import csv
+from datetime import datetime
+import json
+import os
+import sys
```
Comment (marionleborgne) [inline unicorn/scripts/convert_data.py]: `os` and `sys` are unused imports and can be removed.

File: unicorn/scripts/convert_data.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import time
+
+
+
+def convert(dataPath, outputPath, numHeaderRows=None, timestampFormat=None):
+  with open(dataPath, 'r') as infile:
```
Comment (marionleborgne) [inline unicorn/scripts/convert_data.py]: Do you mind adding a docstring to explain what this does?

File: unicorn/scripts/convert_data.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        timestamp = datetime.strptime(row[0], timestampFormat)
+        timestamp = int(time.mktime(timestamp.timetuple()))
+        value = float(row[1])
+        outfile.write(json.dumps([timestamp, value]))
+        outfile.write("\n")
+
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser()
```
Comment (marionleborgne) [inline unicorn/scripts/convert_data.py]: If you think it makes sense, updating the README with an example usage would be good. For example I ran:
```
python convert_data.py <path_to_nupic>/nupic/examples/opf/clients/hotgym/anomaly/one_gym/rec-center-hourly.csv test.json -t "%m/%d/%y %H:%M"
```

File: unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+# ----------------------------------------------------------------------
+
+"""
+Compatibility test of the unicorn_backend.model_runner module
+
+To add data for tests, convert data from NAB using
+unicorn/scripts/convert_data.py and place in the DATA_DIR.
+"""
+
+from datetime import datetime
```
Comment (marionleborgne) [inline unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py]: Unused import that can be removed (`datetime`)

File: unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
```
Comment (marionleborgne) [inline unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py]: `time` can be removed too

File: unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        # Send row for processing
+        mrProcess.stdin.write("%s\n" % (json.dumps(rec),))
+        mrProcess.stdin.flush()
+
+        # Read output row and validate
+        outputRecord = mrProcess.stdout.readline()
+        rowIndex, anomalyLikelihood = json.loads(outputRecord)
+
+        self.assertEqual(rowIndex, i)
+        self.assertAlmostEqual(anomalyLikelihood, results[i][1],
```
Comment (marionleborgne) [inline unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py]: It nothing, but `msg` and everything after can be aligned with the rest :lipstick:

File: unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        data.append(json.loads(line))
+
+    return data
+
+
+  def _testModelRunner(self, name, stats):
+    data = self._load(os.path.join(DATA_DIR, name))
+    results = self._load(os.path.join(RESULTS_DIR, name))
+
+    modelId = uuid.uuid1().hex
```
Comment (marionleborgne) [inline unicorn/tests/py/integration/compatibility_test/model_runner_compatibility_test.py]: I think `modelId` can just be a simple string and does not need to be a UUID. Or is the ID relevant for your tests?

**Comment by marionleborgne** [comment]
@chetan51 Done with the review. I tested the script and test, and added some comments. Thanks!

**Comment by chetan51** [comment]
@marionleborgne Thanks, incorporated your review.

**Comment by marionleborgne** [comment]
Thanks @chetan51

**Pull request merged**